docs/guides/permissions.md

```diff
@@ -4,4 +4,73 @@ title: Permissions
 
 # Permissions
 
-Documentation coming soon
+Permissions can be managed using `Permission` classes. A `Permission` class extends `BasePermission` and has a `has_permission` method. It can be hooked up to a field using the `permission_classes` keyword argument. A simple example looks like this:
+
+```python
+import strawberry
+from strawberry.permission import BasePermission
+from strawberry.types import Info
+
+class IsAuthenticated(BasePermission):
+    message = "User is not authenticated"
+
+    def has_permission(self, source: Any, info: Info, **kwargs) -> bool:
+        return False
+
+@strawberry.type
+class Query:
+    user: str = strawberry.field(permission_classes=[IsAuthenticated])
+```
+
+If the `has_permission` method fails then an error will be raised using the `message` class attribute. See [Dealing with Errors](/docs/guides/errors) for more information.
+
+```json
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "User is not authenticated"
+    }
+  ]
+}
+```
+
+In a simple use case you could obtain all the information you need to execute the `has_permission` check from the `source` and `info` objects.
+
+For more complex cases it might need more information from the Request. This can be provided by extending the `GraphQL` class to provide more context. The `Permission` class can then use that context from the `Info` object.
```

Comment:
Most (maybe all?) of the views already pass the request as part of the context object. Wouldn't it be better to extract what you need in the permission class rather than modifying the context?

Reply:
I didn't know that! I'll update the documentation.

In this case I'm basing the documentation off a solution I used to place a `user` object into the `context` (using the `Authorization` header of the request).